src/uct/tcp/tcp_listener.c

```diff
@@ -34,9 +32,17 @@ static void uct_tcp_listener_conn_req_handler(int fd, void *arg)
 
     ucs_assert(accept_fd != -1);
 
-    ucs_trace("server accepted a connection request from client %s",
-              ucs_sockaddr_str((struct sockaddr*)&client_addr, ip_port_str,
-                               UCS_SOCKADDR_STRING_LEN));
+    if (!ucs_socket_is_connected(accept_fd)) {
```

Comment:
why is this check needed? is it possible accept will return an unconnected socket?

Reply:
the client may disconnect after accept already returned.